### DISCUSSION TITLE
Setting IP Address via Generic Message

### DISCUSSION THREAD

**Discussion by Colt-H** [ottowayi/pycomm3#248] [General]
This is not really a bug, more of an interesting behavior on an edge case that I thought would be fun to share.

The IP Address of most of the Rockwell processors/enet adapters can be changed via generic message. The KB has a few different articles that show doing this, however all of them are communicating over the backplane. For fun I just tried to readdress an ENBT over a network using a generic message and surprisingly it worked even though my script threw an exception. I realized the issue is that where I'm readdressing the device, the path passed to the device when the CIP Driver is called is no longer valid as soon as the message sends. Eventually the receive in socket.py times out. 

The parameters don't appear to change if you try to send the message as an unconnected message, but that could be an issue with my message configuration too. Not too concerned as it works, and I can just catch that exception and try to check the IP address with the newly set IP manually to confirm it changed.

**Comment by ottowayi** [comment]
Yeah, if you're changing the IP address of the device you're directly connected to, you're probably going to have to use a new driver or close/change ip/open again.  If you're changing the IP address of a device further down in your CIP path, then I think unconnected messages will be unaffected, but connected ones would require changing closing, changing the path, and then reopening.
